### ISSUE TITLE
[V9] On odoo-server logs werkzeug KeyError: '40001', odoo failed on request via http

### ISSUE THREAD

**Issue by tejastank** [odoo/odoo#13788]
Impacted versions:
9.0 E
Steps to reproduce:
Server working since many days and sudden issues raise,
I hope its not a critical, 

Current behavior:

```
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/werkzeug/serving.py", line 180, in run_wsgi
    execute(self.server.app)
  File "/usr/lib/python2.7/dist-packages/werkzeug/serving.py", line 168, in execute
    application_iter = app(environ, start_response)
  File "/home/JTA/source/v9e/openerp/service/server.py", line 245, in app
    return self.app(e, s)
  File "/home/JTA/source/v9e/openerp/service/wsgi_server.py", line 184, in application
    return application_unproxied(environ, start_response)
  File "/home/JTA/source/v9e/openerp/service/wsgi_server.py", line 170, in application_unproxied
    result = handler(environ, start_response)
  File "/home/JTA/source/v9e/openerp/http.py", line 1416, in __call__
    return self.dispatch(environ, start_response)
  File "/home/JTA/source/v9e/openerp/http.py", line 1390, in __call__
    return self.app(environ, start_wrapped)
  File "/usr/lib/python2.7/dist-packages/werkzeug/wsgi.py", line 591, in __call__
    return self.app(environ, start_response)
  File "/home/JTA/source/v9e/openerp/http.py", line 1580, in dispatch
    result = ir_http._dispatch()
  File "/home/JTA/source/v9e/openerp/addons/website_sale/models/ir_http.py", line 12, in _dispatch
    return super(IrHttp, self)._dispatch()
  File "/home/JTA/source/v9e/openerp/addons/utm/models/ir_http.py", line 13, in _dispatch
    response = super(ir_http, self)._dispatch()
  File "/home/JTA/source/v9e/openerp/addons/website/models/ir_http.py", line 190, in _dispatch
    resp = super(ir_http, self)._dispatch()
  File "/home/JTA/source/v9e/openerp/addons/web_editor/models/ir_http.py", line 16, in _dispatch
    return super(ir_http, self)._dispatch()
  File "/home/JTA/source/v9e/openerp/addons/base/ir/ir_http.py", line 176, in _dispatch
    return self._handle_exception(e)
  File "/home/JTA/source/v9e/openerp/addons/website/models/ir_http.py", line 239, in _handle_exception
    return super(ir_http, self)._handle_exception(exception)
  File "/home/JTA/source/v9e/openerp/addons/base/ir/ir_http.py", line 147, in _handle_exception
    return request._handle_exception(exception)
  File "/home/JTA/source/v9e/openerp/http.py", line 737, in _handle_exception
    return super(HttpRequest, self)._handle_exception(exception)
  File "/home/JTA/source/v9e/openerp/addons/base/ir/ir_http.py", line 172, in _dispatch
    result = request.dispatch()
  File "/home/JTA/source/v9e/openerp/http.py", line 768, in dispatch
    r = self._call_function(**self.params)
  File "/home/JTA/source/v9e/openerp/http.py", line 315, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/home/JTA/source/v9e/openerp/service/model.py", line 134, in wrapper
    _logger.info("%s, retry %d/%d in %.04f sec..." % (errorcodes.lookup(e.pgcode), tries, MAX_TRIES_ON_CONCURRENCY_FAILURE, wait_time))
  File "/usr/lib/python2.7/dist-packages/psycopg2/errorcodes.py", line 38, in lookup
    return _cache[code]
KeyError: '40001'
```

Expected behavior:

Video/Screenshot link (optional):

**Comment by JKE-be** [comment]
Hello,

This error is returned by psycopg2: https://github.com/psycopg/psycopg2/blob/master/lib/errorcodes.py#L291

You can take a look here, a answer of @odony: https://bugs.launchpad.net/openobject-server/+bug/992525/comments/3

And check that you don't have blocking request / lock in your db...
If you take a look in log, you should see the request that fail.

**Comment by odony** [comment]
@tejastank The problem seems to be that you are running a very old version of psycopg2. The support for this error code was added [back in 2006, in psycopg2 2.0.6](https://github.com/psycopg/psycopg2/commit/43518d067ef5972addc2b0ed94e8520a63e4733d).
Could you check the version you have, and upgrade it, then restart the Odoo server? This should avoid the problem, and activate the proper auto-retry mechanism for transactions.

**Comment by gpothier** [comment]
That could also be because of a bug that was fixed in 2.6.2:
https://github.com/psycopg/psycopg2/issues/382
(this is the latest version at this time)
